text/1026-net-module.rst

```diff
@@ -0,0 +1,223 @@
+::
+
+    Status: Draft
+    Type: Feature
+    Created: 2024-08-14
+    Authors: Scott Trinh <scott@edgedb.com>
+
+=============================
+RFC: EdgeDB Networking Module
+=============================
+
+This RFC outlines the addition of a networking module to EdgeDB’s
+standard library, providing functionality for making HTTP requests and
+sending SMTP messages.
+
+Motivation
+==========
+
+The EdgeDB Auth extension (ext::auth) currently sends emails directly to
+configured SMTP services at certain times in various authentication
+ceremonies and workflows:
+
+-  Sign up with email (sends email verification email)
+-  Password reset (sends password reset email)
+-  Magic link sign up (sends magic link email)
+-  Magic link sign in (sends magic link email)
+
+To allow full customization of emails and provide hooks into
+authentication events generally, we need to support sending webhook
+requests to the application directly. This would enable applications to
+trigger their own logic for sending fully customized and
+internationalized emails, SMS, log analytics, update usage billing
+records, etc.
+
+Specification
+=============
+
+Module Name
+-----------
+
+The new module will be named ``net``.
+
+Utility types
+-------------
+
+Since requests are sent asynchronously, we need a way to track the status of the
+request.
+
+.. code-block:: edgeql
+
+  scalar type net::RequestState extending std::enums<Pending, InProgress, Complete, Failed>;
+
+  scalar type net::RequestFailure extending std::enums<NetworkError, Timeout>;
+
+``net::RequestState`` represents the current asynchronous state of a request.
+The worker process will update the state of the request as it progresses through
+the various states.
+
+``net::RequestFailure`` represents the types of failure that can occur when
+trying to send the request, not a failure response from the server.
```

Comment:
@scotttrinh 

This could be `net::RequestFailureKind` or just `net::FailureKind`, so `ScheduledRequest.failure.failure` would be `ScheduledRequest.failure.kind`.